### ISSUE TITLE
Unexpected spacing for anonymous classes with constructor arguments

### ISSUE THREAD

**Issue by TimWolla** [squizlabs/PHP_CodeSniffer#3200]
**Describe the bug**

When creating an anonymous class while passing constructor arguments phpcs requests a space between `class` and the `(`. I did not expect this and it looks off.

**Code sample**
```php
<?php

namespace foo;

class Foo
{
    protected function foo()
    {
        new class(1) extends Bar { // line 9
        };
    }
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset>
	<file>./test/</file>
	<arg name="extensions" value="php" />
	<arg value="p"/>
	<arg name="basepath" value="."/>

	<rule ref="PSR12"/>
</ruleset>
```

**To reproduce**

1. Create a file called `test.php` with the code sample above.
2. Run `phpcs test.php`
3. See error message displayed

```
FILE: test/test.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 9 | ERROR | [x] Expected 1 space after class keyword; 0 found
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

**Expected behavior**

I expected that error not to be reported, because I believe there should not be a space between `class` and `(`.

**Versions (please complete the following information):**
 - OS: Docker container (Debian based) on Ubuntu host.
 - PHP: 7.4.12
 - PHPCS: 3.5.8 (stable)
 - Standard: PSR12

**Additional context**

[PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) with the PSR2 ruleset expects the space to not be there.

**Comment by TimWolla** [comment]
Looking at this: https://github.com/php-fig/fig-standards/pull/1206 the consensus of the standard appears to be that the space should not be there, even if the current version of PSR-12 does not say anything regarding this topic.

**Comment by gsherwood** [comment]
If the errata is accepted then I'd change the PSR-12 standard included with PHPCS. But right now the PSR-12 standard says:
> Anonymous Classes MUST follow the same guidelines and principles as closures in the above section.

And in the closures section, all the examples clearly show a space between the function keyword and the opening parenthesis of the argument list. So that's all I have to go on until a change is made to clarify things.

**Comment by TimWolla** [comment]
For the record: PHP CS Fixer was changed to match PHP CodeSniffer: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5924
